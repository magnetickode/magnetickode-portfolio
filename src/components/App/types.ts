import { MutableRefObject } from "react";

export interface Props {
  navbarRef: MutableRefObject<HTMLDivElement | null>;
}
