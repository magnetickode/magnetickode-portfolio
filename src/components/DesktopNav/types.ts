export interface Props {
  sticky: boolean;
}

export interface StyledDesktopNavProps {
  sticky: boolean;
}

export interface StyledNavLinkProps {
  sticky: boolean;
}
