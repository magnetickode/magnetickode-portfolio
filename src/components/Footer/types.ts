export interface Props {
  footerState: string;
  footerPosition: string;
}

export interface StyledFooterProps {
  footerState: string;
  footerPosition: string;
}
