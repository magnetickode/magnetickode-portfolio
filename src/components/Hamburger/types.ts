export interface Props {
  opened: boolean;
  toggleHamburger: () => void;
}

export interface StyledHamburgerProps {
  opened: boolean;
}
