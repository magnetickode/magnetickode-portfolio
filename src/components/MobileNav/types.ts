export interface Props {
  collapse: boolean;
  hashRoute: string;
}

export interface StyledMobileNavProps {
  collapse: boolean;
}

export interface StyledNavLinkProps {
  active: boolean;
}
