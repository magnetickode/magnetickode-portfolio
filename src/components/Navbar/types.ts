import { MutableRefObject } from "react";

import { ChangeThemeGen } from "../../store/actions/types";

export interface Props {
  navbarRef?: MutableRefObject<HTMLDivElement | null>;
  isDesktop: boolean;
  stickyNavVisible: boolean;
  changeTheme: ChangeThemeGen;
}

export interface StyledNavbarProps {
  sticky: boolean;
}

export interface LogoProps {
  isDesktop: boolean;
  sticky: boolean;
}
