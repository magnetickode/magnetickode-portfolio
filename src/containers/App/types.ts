import { ScreenSizeChangedGen, ShowStickyNavGen } from "../../store/actions/types";

export interface Props {
  isDesktop: boolean;
  themeIndex: number;
  stickyNavVisible: boolean;
  screenSizeChanged: ScreenSizeChangedGen;
  showStickyNav: ShowStickyNavGen;
}
