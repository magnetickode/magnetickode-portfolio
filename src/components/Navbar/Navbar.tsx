import React, { useState } from "react";
import styled from "styled-components";

import Hamburger from "../Hamburger";
import { Props, StyledNavbarProps, LogoProps } from "./types";

const StyledNavbar = styled.div<StyledNavbarProps>`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  transition: all 0.8s ease-in-out;

  & > div {
    width: 110rem;
    max-width: 100%;
    padding: 1.5rem 2rem;
    margin: auto;
    display: flex;
    justify-content: space-between;
    align-items: center;
    box-sizing: border-box;
  }

  @media (min-width: 720px) {
    & > div {
      padding: 2rem;
    }
  }

  ${({ sticky, theme }) =>
    sticky &&
    `   
      background: ${theme.textColor};  
  `}
`;

const Logo = styled.div<LogoProps>`
  display: flex;
  justify-content: center;
  align-items: center;
  width: ${({ isDesktop }) => (isDesktop ? "6rem" : "4.5rem")};
  height: ${({ isDesktop }) => (isDesktop ? "6rem" : "4.5rem")};
  color: ${({ sticky, theme }) => (sticky ? theme.primaryColor : theme.textColor)};
  border: 0.4rem solid
    ${({ sticky, theme }) => (sticky ? theme.primaryColor : theme.textColor)};
  border-radius: ${({ isDesktop }) => (isDesktop ? "6rem" : "4.5rem")};
  font-family: "Pacifico", cursive;
  font-size: ${({ isDesktop }) => (isDesktop ? "3.2rem" : "2.6rem")};
  user-select: none;
  cursor: pointer;
`;

const Navbar: React.FC<Props> = ({
  navbarRef,
  isDesktop,
  stickyNavVisible,
  changeTheme
}) => {
  const [hamburgerOpened, setHamburger] = useState(false);

  const toggleHamburger = () => setHamburger(prevState => !prevState);

  return (
    <StyledNavbar data-test="NavbarComponent" ref={navbarRef} sticky={stickyNavVisible}>
      <div>
        <Logo isDesktop={isDesktop} sticky={stickyNavVisible} onClick={changeTheme}>
          mk
        </Logo>
        {isDesktop ? (
          <p>Desktop</p>
        ) : (
          <>
            <Hamburger
              data-test="Hamburger"
              opened={hamburgerOpened}
              sticky={stickyNavVisible}
              toggleHamburger={toggleHamburger}
            />
          </>
        )}
      </div>
    </StyledNavbar>
  );
};

export default Navbar;
