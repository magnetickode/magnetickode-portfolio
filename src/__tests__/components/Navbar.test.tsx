import React from "react";
import { shallow, ShallowWrapper } from "enzyme";

import Navbar from "../../components/Navbar";
import { changeTheme } from "../../store/actions/mainActions";
import { Props } from "../../components/Navbar/types";

const props: Props = {
  isDesktop: false,
  stickyNavVisible: false,
  changeTheme
};

describe("screen size mobile", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Navbar {...props} />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders without error", () => {
    const NavbarComponent = wrapper.find("[data-test='NavbarComponent']");
    expect(NavbarComponent.length).toBe(1);
  });

  it("renders Hamburger component", () => {
    const Hamburger = wrapper.find("[data-test='Hamburger']");
    expect(Hamburger.length).toBe(1);
  });
});

describe("screen size desktop", () => {
  const desktopProps: Props = {
    ...props,
    isDesktop: true
  };

  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Navbar {...desktopProps} />);
  });

  it("does not render Hamburger component", () => {
    const Hamburger = wrapper.find("[data-test='Hamburger']");
    expect(Hamburger.length).toBe(0);
  });
});
