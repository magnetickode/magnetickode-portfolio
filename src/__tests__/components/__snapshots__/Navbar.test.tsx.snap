// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen size mobile matches snapshot 1`] = `
<styled.div
  data-test="NavbarComponent"
  sticky={false}
>
  <div>
    <styled.div
      isDesktop={false}
      onClick={[Function]}
      sticky={false}
    >
      mk
    </styled.div>
    <Hamburger
      data-test="Hamburger"
      opened={false}
      sticky={false}
      toggleHamburger={[Function]}
    />
  </div>
</styled.div>
`;
